src/axom/core/tests/core_flatmap.hpp

```diff
@@ -670,3 +677,66 @@ AXOM_TYPED_TEST(core_flatmap, range_for_loop_write)
     EXPECT_EQ(test_map[key], expected_value);
   }
 }
+#if defined(AXOM_USE_HIP) || defined(AXOM_USE_CUDA)
+
+AXOM_TYPED_TEST(core_flatmap, copy_host_device)
+{
+  using MapType = typename TestFixture::MapType;
+
+  using HostExec = typename TestFixture::HostExec;
+  using DeviceExec = typename TestFixture::DeviceExec;
+
+  const int host_alloc_id = axom::execution_space<HostExec>::allocatorID();
+  const int device_alloc_id = axom::execution_space<DeviceExec>::allocatorID();
```

Comment:
Longer term, should we have axom::execution_space<ExecSpace>::allocator() that returns an Allocator? That would get rid of some of these lines where Allocator needs to be constructed from the integer id.